code/modules/networks/computer3/bank_rec.dm

```diff
@@ -0,0 +1,1050 @@
+							//If you have a special role and you're on the manifest everything is probably normal
+	)
+
+	// printer stuff
+	var/tmp/connected = 0
+	var/tmp/server_netid = null
+	var/tmp/potential_server_netid = null
+	var/tmp/selected_printer = null
+	var/tmp/list/known_printers = list()
+	var/tmp/printer_status = "???"
+
+	var/setup_logdump_name = "banklog"
+	var/setup_mail_freq = FREQ_PDA
+
+/datum/computer/file/terminal_program/bank_records/initialize()
+	if (..())
+		return TRUE
+
+	src.record_database = data_core.general
+	src.master.temp = null
+	src.menu = MENU_MAIN
+	src.field_input = 0
+
+	src.log_string += "<br><b>LOGIN:</b> [src.authenticated]"
+
+	src.print_text(src.mainmenu_text())
+
+/datum/computer/file/terminal_program/bank_records/input_text(text)
+	if(..())
+		return
+
+	var/list/command_list = parse_string(text)
+	var/command = command_list[1]
+	command_list -= command_list[1]
+
+	switch(menu)
+		if(MENU_MAIN)
+			switch(text2num_safe(command))
+				if(MENU_MAIN_OPT_QUIT)
+					src.print_text("Quitting...")
+					src.master.unload_program(src)
+					return
+				if(MENU_MAIN_OPT_INDEX)
+					src.record_database = data_core.general
+					src.menu = MENU_INDEX
+					src.print_index()
+				if(MENU_MAIN_OPT_SEARCH)
+					src.print_text("Please enter target name, ID, or rank.")
+					src.menu = MENU_SEARCH_INPUT
+					return
+				if(MENU_MAIN_OPT_BUDGET)
+					src.print_budget()
+					src.menu = MENU_STATION_BUDGET
+					return
+
+		if(MENU_INDEX)
+			var/index_number = round(max(text2num_safe(command), 0))
+			if (index_number == 0)
+				src.menu = MENU_MAIN
+				src.master.temp = null
+				src.print_text(mainmenu_text())
+				return
+
+			if (!istype(record_database) || index_number > record_database.records.len)
+				src.print_text("Invalid record.")
+				return
+
+			var/datum/db_record/check = src.record_database.records[index_number]
+			if (!check || !istype(check))
+				src.print_text("<b>Error:</b> Record Data Invalid.")
+				return
+
+			src.active_general = check
+			if (data_core.general.has_record(check))
+				src.active_bank = data_core.bank.find_record("id", src.active_general["id"])
+				if (!src.active_bank)
+					data_core.bank.find_record("name", src.active_general["name"])
+
+			src.log_string += "<br>Log loaded: [src.active_general["id"]]"
+
+			if (src.print_active_record())
+				src.menu = MENU_IN_RECORD
+			return
+		if(MENU_IN_RECORD)
+			switch(lowertext(command))
+				if ("r")
+					src.print_active_record()
+					return
+				if ("d")
+					src.print_text("Are you sure? (Y/N)")
+					src.field_input = FIELDNUM_DELETE
+					src.menu = MENU_FIELD_INPUT
+					return
+				if ("p")
+					if ((src.connected && src.selected_printer) && !src.network_print())
+						src.print_text("Print instruction sent.")
+					else
+						if (local_print())
+							src.print_text("<b>Error:</b> No printer detected.")
+						else
+							src.print_text("Print instruction sent.")
+					return
+				if (FIELDNUM_NEWREC)
```

Comment:
I'm confused when this is supposed to happen? If I don't have a record then my name won't be in the list and I'll never get here, and if I scan myself with a recordtrak to give myself a record then I get a bank record created automatically.

Reply:
this case is when you have a general record but no bank record, which can happen if:
1. someone deletes your bank record, but not general record
2. someone creates a general record, but there isn't a bank record created yet (i.e. sec makes a record for you manually)